app/validators/blocks/list_collector_validator.py

```diff
@@ -67,19 +69,32 @@ def validate_list_collector_answer_ids(self, block):
         """
         - Ensure that answer_ids on add and edit blocks match between all blocks that populate a single list.
         - Enforce the same answer_ids on add and edit sub-blocks
+        - Ensure that that child block answer_ids are not used elsewhere in the schema that's not another list collector
         """
-        add_answer_ids = self.questionnaire_schema.get_all_answer_ids(
-            block["add_block"]["id"]
-        )
-        edit_answer_ids = self.questionnaire_schema.get_all_answer_ids(
-            block["edit_block"]["id"]
-        )
+        list_block_answer_ids = {
```

Comment:
Now that you have 3 dict comprehensions exactly like this, might be worth a shared method like `get_list_block_answer_ids(block_id: str)`

Reply:
Great shout - thanks. Have created a common method for this in questionnaire_schema as `get_list_collector_answer_ids_by_child_block` to handle these 3 areas 👍